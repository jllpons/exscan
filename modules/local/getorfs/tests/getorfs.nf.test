nextflow_process {

    name "Test Process GETORFS"
    script "../main.nf"
    process "GETORFS"

    test("Testing process GETORFS on *E. coli* LexA...") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/fasta/lexA_Ecoli.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}

