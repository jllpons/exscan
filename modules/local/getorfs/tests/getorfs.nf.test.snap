{
    "Testing process GETORFS on *E. coli* LexA...": {
        "content": [
            {
                "0": [
                    "lexA_Ecoli.orfs.fasta:md5,ff5cc0755615e47a7986a02dde8f1400"
                ],
                "1": [
                    "versions.yml:md5,69e2bbb0aeaec7c42a1ab0b87b7920f6"
                ],
                "orfs": [
                    "lexA_Ecoli.orfs.fasta:md5,ff5cc0755615e47a7986a02dde8f1400"
                ],
                "versions": [
                    "versions.yml:md5,69e2bbb0aeaec7c42a1ab0b87b7920f6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-28T12:10:01.145835"
    }
}